### PR TITLE
customizable look and feel for portal

### PR DESCRIPTION
**Pull request by baldarn**
# Comprehensive Summary of your change

This PR aims to let a user configure the `look and feel` from the `harbor.yml` file

# Issue being fixed
Fixes https://github.com/goharbor/harbor/issues/20847

Please indicate you've done the following:
- [ ] Well Written Title and Summary of the PR
- [ ] Label the PR as needed. "release-note/ignore-for-release, release-note/new-feature, release-note/update, release-note/enhancement, release-note/community, release-note/breaking-change, release-note/docs, release-note/infra, release-note/deprecation"
- [x] Accepted the DCO. Commits without the DCO will delay acceptance.
- [ ] Made sure tests are passing and test coverage is added if needed.
- [ ] Considered the docs impact and opened a new docs issue or PR with docs changes if needed in [website repository](https://github.com/goharbor/website).

### CODE REVIEW

**Comment by baldarn** [comment]
hi, any updates @OrlinVasilev ? is this pointing in a good direction?

**Comment by OrlinVasilev** [comment]
@Vad1mo @wy65701436 can you check please

**Review by Vad1mo — COMMENTED** [review]
Do I understand the code correctly? 
This change does only kicks in on the installer and copies the look_and_feel tree in config.json.

**Comment by baldarn** [comment]
> Do I understand the code correctly? This change does only kicks in on the installer and copies the look_and_feel tree in config.json.

yep, this also allows you to bind images to the `portal` docker container.

File: make/photon/prepare/utils/configs.py [hunk 1]
```diff
@@ -347,6 +347,9 @@ def parse_yaml_config(config_file_path, with_trivy):
     core_config = configs.get('core')
     config_dict['core'] = Core(core_config or {})
 
+    # look and feel configs
+    config_dict['look_and_feel'] = configs.get('look_and_feel') or {}
```
Comment (josegomezr) [inline make/photon/prepare/utils/configs.py:351]: Did you mean?

> "Assign the value of `configs.get('look_and_feel')` and when missing fallback to `{}`"

```suggestion
    config_dict['look_and_feel'] = configs.get('look_and_feel', {})
```

File: make/photon/prepare/utils/portal.py [hunk 2]
```diff
@@ -19,3 +21,11 @@ def prepare_portal(config_dict):
         gid=DEFAULT_GID,
         strong_ssl_ciphers=config_dict['strong_ssl_ciphers']
         )
+
+    # Render Portal setting config
+    if config_dict['look_and_feel']:
```
Comment (josegomezr) [inline make/photon/prepare/utils/portal.py:26]: Did you mean?

> If config_dict has something stored in `'look_and_feel'` key: then render the jinja remplate`

```suggestion
    if 'look_and_feel' in config_dict:
```
or:

```suggestion
    if config_dict.get('look_and_feel', False):
```

**Comment by baldarn** [comment]
:heart: @josegomezr

**Comment by OrlinVasilev** [comment]
@baldarn are you still up for this?

**Comment by baldarn** [comment]
@OrlinVasilev yea, to go on I would like to have a review. I'm really not into this logic. is this something that have sense? XD

**Pull request closed without merging**